### PR TITLE
Geod 498 make 4character id field readonly

### PR DESCRIPTION
**Pull request by heyagosper**

### CODE REVIEW

File: src/client/app/shared/toolbar/toolbar.component.ts [hunk 1]
```diff
@@ -70,8 +70,17 @@ export class ToolbarComponent implements OnInit, OnDestroy {
...
+     * save is always enabled for new site because everything is optional
+     * save is disabled if nothing has in the form has changed or the user is not allowed to edit this site
+     */
     public isSaveDisabled(): boolean {
-        return !(this.hasAuthorityToEditSite() && this.isFormDirty());
+        let isSaveDisabled = false;
+        if (this.siteId !== 'newSite') {
+            isSaveDisabled = !this.isFormDirty() || !this.hasAuthorityToEditSite();
+        }
+        return isSaveDisabled;
```
Comment (lbodor) [inline src/client/app/shared/toolbar/toolbar.component.ts]: I think that save should be disabled for new sites when the form is not dirty. Why save an empty form?

**Comment by lbodor** [comment]
:+1:

**Comment by lbodor** [comment]
:+1:

File: src/client/app/shared/dynamic-form-fields/text-input.component.html [hunk 2]
```diff
@@ -6,6 +6,7 @@
     <div class="text-input" class="col-md-5 col-sm-6 col-xs-6 col-xxs-12">
          <input (keyup.enter)="ignoreEvent($event);"
              type="text"
+             readonly="{{readonly}}"
```
Comment (oehm-smith) [inline src/client/app/shared/dynamic-form-fields/text-input.component.html]: Square brackets instead.  Yesterday when you had a similar thing for invalid(), the {{}} doesn't seem to refresh all the tiem.

**Pull request merged**